pkg/readiness/objset.go

```diff
@@ -27,9 +27,39 @@ type objKey struct {
 	namespacedName types.NamespacedName
 }
 
-type objSet map[objKey]struct{}
-
 func (k objKey) String() string {
 	return fmt.Sprintf("%s [%s]", k.namespacedName.String(), k.gvk.String())
+}
+
+type objData struct {
+	retries int
+}
+
+type objDataMutator func(*objData) *objData
+
+func setRetriesFromFlag(o *objData) *objData {
+	o.retries = *readinessRetries
+	return o
+}
+
+func mutatedObjData(muts ...objDataMutator) objData {
+	out := objData{}
+	for _, m := range muts {
+		m(&out)
+	}
+	return out
+}
+
+type objSet map[objKey]objData
 
+func (o objSet) decrementOrDelete(key objKey) bool {
+	val := o[key]
```

Comment:
what happens if the key is missing?

Reply:
It would ultimately return a zero-valued `objData`, which would trigger the `else` clause below.  The `delete` would do nothing (key doesn't exist in map) and then the function would return `true`.  

This really _isn't_ a break from the previous behavior, which was a call to `delete`.  That said, it's not as clear as it could be.  I can make it `val, ok := o[key]` and return `true` (given the non-existent record is not decremented).

That said, I think this is poking at a question that's bigger than this function: how should `objectTracker` deal with bogus keys?  With or without my change, those keys end up in `t.cancelled`.  Is that a problem?